src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java

```diff
@@ -0,0 +1,410 @@
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_PROCESSING_RESPONSE_FILES;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_PROCESSING_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ArmResponseFilesProcessorImpl implements ArmResponseFilesProcessor {
+
+    public static final String ARM_RESPONSE_FILE_EXTENSION = ".rsp";
+    public static final String ARM_INPUT_UPLOAD_FILENAME_KEY = "iu";
+    public static final String ARM_CREATE_RECORD_FILENAME_KEY = "cr";
+    public static final String ARM_UPLOAD_FILE_FILENAME_KEY = "uf";
+    public static final String ARM_RESPONSE_SUCCESS_STATUS_CODE = "1";
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final ArmDataManagementApi armDataManagementApi;
+    private final FileOperationService fileOperationService;
+    private final ArmDataManagementConfiguration armDataManagementConfiguration;
+    private final ObjectMapper objectMapper;
+    private final UserIdentity userIdentity;
+
+    @Transactional
+    @Override
+    public void processResponseFiles() {
+        startProcessingResponseFiles();
+    }
+
+    private void startProcessingResponseFiles() {
+        // Fetch All records from external Object Directory table with external_location_type as 'ARM' and status with 'ARM dropzone'.
+        ExternalLocationTypeEntity armLocation = externalLocationTypeRepository.getReferenceById(ExternalLocationTypeEnum.ARM.getId());
+
+        ObjectRecordStatusEntity armDropZoneStatus = objectRecordStatusRepository.getReferenceById(ARM_DROP_ZONE.getId());
+        ObjectRecordStatusEntity armProcessingResponseFilesStatus = objectRecordStatusRepository.getReferenceById(ARM_PROCESSING_RESPONSE_FILES.getId());
+
+        List<ExternalObjectDirectoryEntity> dataSentToArm =
+            externalObjectDirectoryRepository.findByExternalLocationTypeAndObjectStatus(armLocation, armDropZoneStatus);
+
+        for (ExternalObjectDirectoryEntity externalObjectDirectory : dataSentToArm) {
+            updateExternalObjectDirectory(externalObjectDirectory, armProcessingResponseFilesStatus);
+        }
+        for (ExternalObjectDirectoryEntity externalObjectDirectory : dataSentToArm) {
+            try {
+                findInputUploadFile(externalObjectDirectory);
+            } catch (Exception e) {
+                log.error("Unable to process response files for external object directory {}", e.getMessage());
+                updateExternalObjectDirectoryStatusAndVerificationAttempt(externalObjectDirectory, armDropZoneStatus);
+            }
+        }
+    }
+
+    private void findInputUploadFile(ExternalObjectDirectoryEntity externalObjectDirectory) {
```

Comment:
This method isn't returning anything, yet us called "find....", can you rename the method to be descriptive as to what its doing?